python/utils/bootstrap_utilities.py

```diff
@@ -391,3 +392,39 @@ def handle_error(data, proxy=None):
             proxy.close()
         except Exception:
             pass
+
+
+def execute_pre_initialization_hook():
+    """
+    Executes the 'hook_pre_initialization' hook.
+
+    This function attempts to execute the hook specified in the environment variable
+    'SGTK_HOOK_PRE_INITIALIZATION'. If the variable is defined, it dynamically loads
+    and executes the hook from the specified path. If the variable is not defined,
+    it falls back to dynamically loading the default hook located in the 'hooks'
+    directory of the project.
+
+    The purpose of this hook is to perform early initialization tasks, such as importing
+    necessary libraries, before PySide6 or other dependencies are loaded. This ensures
+    a stable environment and avoids potential conflicts caused by Qt initialization issues
+    or version mismatches.
+    """
+
+    # Retrieve the hook path from the environment variable
+    hook_path = os.environ.get("SGTK_HOOK_PRE_INITIALIZATION")
+
+    # Check if the hook file exists, so if the environment
+    # variable is not set, use the default hook path
+    if not hook_path or not os.path.exists(hook_path):
+        hook_path = os.path.join(
+            os.path.dirname(__file__), "..", "..", "hooks", "pre_initialization.py"
+        )
```

Comment:
I think we need a log if the setting is set but not a valid path
```suggestion
    if hook_path and not os.path.exists(hook_path):
        logger.error("No such file %s; reverting to hook's default value")
        hook_path = None

     if not hook_path:
        hook_path = os.path.join(
            os.path.dirname(__file__), "..", "..", "hooks", "pre_initialization.py"
        )
```

We probably need a log line a bit better than my suggestion